OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -21,7 +21,19 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Attach this to a unit to enable dynamic upgrades by warheads, experience, crates, support powers, etc.")]
 	public class UpgradeManagerInfo : ITraitInfo, Requires<IUpgradableInfo>
 	{
-		public object Create(ActorInitializer init) { return new UpgradeManager(init); }
+		[UpgradeGrantedReference]
+		[Desc("Upgrades that can be granted/revoked from Lua scripts.")]
+		public readonly HashSet<string> Scriptable = new HashSet<string>();
+
+		[UpgradeUsedReference]
+		[Desc("Upgrades not to check whether used.")]
+		public readonly HashSet<string> IgnoredAsUnused = new HashSet<string>();
```

Comment:
No, right now we lint-check everything. Just in a DYI way for `GainsExperience` and `Pluggable`.
I like the part of this PR about streamlining the checking by adding dictionary support to `LintExts.GetFieldValues()`.
None of that has any relation to the hashset above though, and what it's there to do.